databricks/koalas/series.py

```diff
@@ -416,6 +416,136 @@ def is_unique(self):
+
+        >>> ks.Series([True, True, None]).all()
+        True
+
+        >>> ks.Series([True, False, None]).all()
+        False
+
+        >>> ks.Series([]).all()
+        True
+
+        >>> ks.Series([np.nan]).all()
+        True
+        """
+
+        if axis not in [0, 'index']:
+            raise ValueError('axis should be either 0 or "index" currently.')
+
+        sdf = self._kdf._sdf.select(self._scol)
+        col = self._scol
+
+        # any and every was added as of Spark 3.0
+        # ret = sdf.select(F.expr("every(CAST(`%s` AS BOOLEAN))" % sdf.columns[0])).collect()[0][0]
+        # if ret is None:
+        #     return True
+        # else:
+        #     return ret
+
+        # Note that we're ignoring `None`s here for now.
+        # Here we check the count without nulls is the same with the number of `True`s
+        # in order to mimic `all`.
+        return sdf.select(
+            (F.count(col)) ==
+            (F.count(F.when(col.cast('boolean'), 1).otherwise(None)))
+        ).collect()[0][0]
+
+    # TODO: axis, skipna, and many arguments should be implemented.
+    def any(self, axis: int = 0) -> bool:
+        """
+        Return whether any element is True.
+
+        Returns False unless there at least one element within a series that is
+        True or equivalent (e.g. non-zero or non-empty).
+
+        Parameters
+        ----------
+        axis : {0 or 'index'}, default 0
+            Indicate which axis or axes should be reduced.
+
+            * 0 / 'index' : reduce the index, return a Series whose index is the
+              original column labels.
+
+        Examples
+        --------
+
+        **Series**
+
+        For Series input, the output is a scalar indicating whether any element
+        is True.
+
+        >>> ks.Series([False, False]).any()
+        False
+
+        >>> ks.Series([True, False]).any()
+        True
+
+        >>> ks.Series([0, 0]).any()
+        False
+
+        >>> ks.Series([0, 1, 2]).any()
+        True
+
+        >>> ks.Series([False, False, None]).any()
+        False
+
+        >>> ks.Series([True, False, None]).any()
+        True
+
+        >>> ks.Series([]).any()
+        False
+
+        >>> ks.Series([np.nan]).any()
+        False
+        """
+
+        if axis not in [0, 'index']:
+            raise ValueError('axis should be either 0 or "index" currently.')
+
+        sdf = self._kdf._sdf.select(self._scol)
+        col = self._scol
+
+        # any and every was added as of Spark 3.0
+        # ret = sdf.select(F.expr("any(CAST(`%s` AS BOOLEAN))" % sdf.columns[0])).collect()[0][0]
+        # if ret is None:
+        #     return False
+        # else:
+        #     return ret
+
+        # Note that we're ignoring `None`s here for now.
+        # Here we check if the count of `True`s is more than one in order to mimic `any`.
+        return sdf.select(
+            (F.count(F.when(col.cast('boolean'), 1).otherwise(None)) >= 1)
+        ).collect()[0][0]
```

Comment:
@hvanhovell do you remember this discussion about rewriting for `any` and `every`? (IIRC you suggested max/min approach but the hole was `None` handling). Can you quickly check if this rewriting makes sense to you?

Here I need to mimic both here to mimic Pandas' with rewriting since `any` and `every` only exists in the master.

Reply:
I can't entirely remember that comment. I did found an e-mail thread about this, and there I state the opposite: `ANY(col)` can be safely rewritten into `MAX(col)` you may need to add a coalesce if we need to return `false` for an empty dataset.